### PR TITLE
[SPARK-47725][INFRA] Set up the CI for pyspark-connect package

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to set up a scheduled job for `pyspark-connect` package. The CI:
1. Build Spark
2. Package `pyspark-connect` with test cases
3. Remove `python/lib/pyspark.zip` and `python/lib/py4j.zip` to make sure we don't use JVM
4. Run the test cases packaged together within `pyspark-connect`.

### Why are the changes needed?

In order to make sure on the feature coverage in `pyspark-connect`.

### Does this PR introduce _any_ user-facing change?

No, test-only.

### How was this patch tested?

Manually tested in my fork, https://github.com/HyukjinKwon/spark/actions/runs/8598881063

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8548784419

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8548832973/job/23423174130

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8548849051

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8549142728

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8549163391

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8549387447

**Comment by HyukjinKwon** [comment]
https://github.com/HyukjinKwon/spark/actions/runs/8549624298

File: python/pyspark/core/context.py [hunk 1]
```diff
@@ -145,7 +145,7 @@ class SparkContext:
 
     Examples
     --------
-    >>> from pyspark.core.context import SparkContext
+    >>> from pyspark.context import SparkContext
```
Comment (HyukjinKwon) [inline python/pyspark/core/context.py:148]:
```suggestion
    >>> from pyspark.core.context import SparkContext
```

**Comment by 09306677806** [comment]
Withdrawing only my physical body in my branch and my bank, my physical
body and my ego, if it is through the real system, it will be ok, and all
the fingerprints should be physically real, and wash your hands with a
strong cleaner before handing over to the main bank.

ShahrzadMahro

در تاریخ پنجشنبه ۴ آوریل ۲۰۲۴،‏ ۰۶:۴۹ Hyukjin Kwon ***@***.***>
نوشت:

File: .github/workflows/build_python_connect.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+name: Build / Spark Connect Python-only (master, Python 3.11)
+
+on:
+  push:
+    branches:
+    - '**'
+
+jobs:
+  # Build: build Spark and run the tests for specified modules using maven
```
Comment (HyukjinKwon) [inline .github/workflows/build_python_connect.yml:28]:
```suggestion
  # Build: build Spark and run the tests for specified modules using SBT
```

File: .github/workflows/build_python_connect.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        env:
+          SPARK_TESTING: 1
+        run: |
+          cd python
+          python packaging/connect/setup.py sdist
+          cd dist
+          pip install pyspark-connect-*.tar.gz
+      - name: Run tests
+        env:
+          SPARK_CONNECT_TESTING_REMOTE: sc://localhost
```
Comment (HyukjinKwon) [inline .github/workflows/build_python_connect.yml:77]:
```suggestion
          SPARK_CONNECT_TESTING_REMOTE: sc://localhost
          SPARK_TESTING: 1
```

File: python/pyspark/sql/tests/connect/test_connect_session.py [hunk 4]
```diff
@@ -248,7 +250,7 @@ def setUp(self) -> None:
             .config("integer", 1)
             .config("boolean", False)
             .appName(self.__class__.__name__)
-            .remote("local[4]")
+            .remote(os.environ.get("SPARK_CONNECT_TESTING_REMOTE", "local[4]"))
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/connect/test_connect_session.py:253]:
```suggestion
            .remote("local[4]")
```

File: python/pyspark/sql/tests/test_catalog.py [hunk 5]
```diff
@@ -73,6 +73,9 @@ def test_list_tables(self):
             spark.sql("CREATE DATABASE some_db")
             with self.table("tab1", "some_db.tab2", "tab3_via_catalog"):
                 with self.tempView("temp_tab"):
+                    # Make sure tables are all dropped.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_catalog.py:76]:
```suggestion
```

File: python/pyspark/sql/tests/test_catalog.py [hunk 6]
```diff
@@ -73,6 +73,9 @@ def test_list_tables(self):
...
+                    for t in spark.catalog.listTables():
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_catalog.py:77]:
```suggestion
```

File: python/pyspark/sql/tests/test_catalog.py [hunk 7]
```diff
@@ -73,6 +73,9 @@ def test_list_tables(self):
...
+                        self.spark.sql("DROP TABLE IF EXISTS %s" % t.name)
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_catalog.py:78]:
```suggestion
```

**Comment by HyukjinKwon** [comment]
Test results: https://github.com/HyukjinKwon/spark/actions/runs/8598881063

Should be ready to go, cc @zhengruifeng @dongjoon-hyun @ueshin

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Nice. Thank you, @HyukjinKwon .

**Pull request closed without merging**